py-polars/polars/dataframe/frame.py

```diff
@@ -2551,18 +2551,18 @@ def write_csv(
         >>> df.write_csv(path, separator=",")
 
         """
-        if len(separator) > 1:
+        if len(separator) != 1:
             raise ValueError("only single byte separator is allowed")
-        elif len(quote) > 1:
+        if len(quote) != 1:
             raise ValueError("only single byte quote char is allowed")
-        elif null_value == "":
+        if not null_value:
```

Comment:
Can this keep the original `""` check?

Reply:
I prefer to check in this way as it's more pythonic to check for truthiness instead of a specific type.
For example in my case you could (potentially) pass in `[]` or `0` or ... Basically anything that resolved `bool(null_value)` to `False`. In the case `null_value == ""` you check specifically for an empty string, whereas `not null_value` checks for any Falsy value.

In the former case, it will fail with an error within Rust because of the wrong type. In the latter it will be replaced by None.

There is something to say for both options and I'll defer to your judgement.
Please do consider the pros and cons and let me know how you would like to have it resolved.

Reply:
Yes, but I believe an empty string evaluates to True, so you now changed the semantics of this code.

Reply:
Nope :-), I did not change anything 😀:

```python
>>> for tst in ["", 0, [], {}, "0", False, -1]:
...     print(tst, type(tst), bool(tst))
...
 <class 'str'> False
0 <class 'int'> False
[] <class 'list'> False
{} <class 'dict'> False
0 <class 'str'> True
False <class 'bool'> False
-1 <class 'int'> True
```

Like you know rust, I know Python 😁...

Would that change your mind?

Reply:
My bad! All good. :+1: